pkg/monitor/monitor/monitor_manager.go

```diff
@@ -389,37 +396,15 @@ func (m *MonitorManager) removeIngressIfExist(monitor *v1alpha1.TidbMonitor, nam
 	return m.deps.TypedControl.Delete(monitor, ingress)
 }
 
-func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef, monitor *v1alpha1.TidbMonitor) error {
-	tc, err := m.deps.Clientset.PingcapV1alpha1().TidbClusters(tcRef.Namespace).Get(tcRef.Name, metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	var mergePatch []byte
-	if tcRef != nil {
-		grafanaEnabled := true
-		if monitor.Spec.Grafana == nil {
-			grafanaEnabled = false
-		}
-		mergePatch, err = json.Marshal(map[string]interface{}{
-			"status": map[string]interface{}{
-				"monitor": map[string]interface{}{
-					"name":           monitor.Name,
-					"namespace":      monitor.Namespace,
-					"grafanaEnabled": grafanaEnabled,
-				},
-			},
-		})
-	} else {
-		mergePatch, err = json.Marshal(map[string]interface{}{
-			"status": map[string]interface{}{
-				"monitor": nil,
-			},
-		})
-	}
-	if err != nil {
-		return err
+func (m *MonitorManager) patchTidbClusterStatus(tc *v1alpha1.TidbCluster, monitor *v1alpha1.TidbMonitor) error {
+	var err error
+	grafanaEnabled := true
+	if monitor.Spec.Grafana == nil {
+		grafanaEnabled = false
 	}
-	_, err = m.deps.Clientset.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
+	tc.Status.Monitor = &v1alpha1.TidbMonitorRef{Name: monitor.Name,
+		Namespace: monitor.Namespace, GrafanaEnabled: grafanaEnabled}
+	_, err = m.deps.TiDBClusterControl.UpdateTidbCluster(tc, nil, nil)
```

Comment:
Please keep the original `Patch` here, this may overwrite the status that updated in the TidbCluster controller.

Reply:
Patch method is not well for unit test.